.github/workflows/typespec-validation-all.yaml

```diff
@@ -51,7 +55,7 @@ jobs:
 
       - uses: actions/checkout@v4
         with:
-          fetch-depth: 2
+          ref: ${{ matrix.ref != 'default' && matrix.ref || '' }}
```

Comment:
If `ref` is set to empty string, it defaults to checking out the context associated with the trigger event, which is usually what you want.